spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/AbstractBinder.java

```diff
@@ -166,107 +157,21 @@ protected final String groupedName(String name, String group) {
 	}
 
 	protected final MessageValues serializePayloadIfNecessary(Message<?> message) {
-		Object originalPayload = message.getPayload();
-		Object originalContentType = message.getHeaders().get(MessageHeaders.CONTENT_TYPE);
-
-		// Pass content type as String since some transport adapters will exclude
-		// CONTENT_TYPE Header otherwise
-		Object contentType = JavaClassMimeTypeConversion
-				.mimeTypeFromObject(originalPayload, ObjectUtils.nullSafeToString(originalContentType)).toString();
-		Object payload = serializePayloadIfNecessary(originalPayload);
-		MessageValues messageValues = new MessageValues(message);
-		messageValues.setPayload(payload);
-		messageValues.put(MessageHeaders.CONTENT_TYPE, contentType);
-		if (originalContentType != null && !originalContentType.toString().equals(contentType.toString())) {
-			messageValues.put(BinderHeaders.BINDER_ORIGINAL_CONTENT_TYPE, originalContentType.toString());
-		}
-		return messageValues;
+		return MessageSerializationUtils.serializePayload(message, this.codec);
 	}
 
 	protected final byte[] serializePayloadIfNecessary(Object originalPayload) {
-		if (originalPayload instanceof byte[]) {
-			return (byte[]) originalPayload;
-		}
-		else {
-			ByteArrayOutputStream bos = new ByteArrayOutputStream();
-			try {
-				if (originalPayload instanceof String) {
-					return ((String) originalPayload).getBytes("UTF-8");
-				}
-				this.codec.encode(originalPayload, bos);
-				return bos.toByteArray();
-			}
-			catch (IOException e) {
-				throw new SerializationFailedException(
-						"unable to serialize payload [" + originalPayload.getClass().getName() + "]", e);
-			}
-		}
+		return MessageSerializationUtils.serializePayload(originalPayload, this.codec);
 	}
 
 	protected final MessageValues deserializePayloadIfNecessary(Message<?> message) {
-		return deserializePayloadIfNecessary(new MessageValues(message));
+		return MessageSerializationUtils.deserializePayload(new MessageValues(message), this.contentTypeResolver,
+				this.payloadTypeCache, this.codec);
```

Comment:
This `payloadTypeCache` must go to the `MessageSerializationUtils` directly.
I see you have something similar in the `JavaClassMimeTypeUtils` for the `mimeTypesCache`. 
So, why don't be consistent?